### PR TITLE
[Core] [Preferences] Modernize for loop in DlgSettingsEditor.cpp

### PR DESCRIPTION
**Pull request by CalligaroV**
Related to #11969

According to suggestions, in `DlgSettingsEditor.cpp` the for loops have been modernized to range-for loops

### CODE REVIEW

File: src/Gui/PreferencePages/DlgSettingsEditor.cpp [hunk 1]
```diff
@@ -163,9 +163,10 @@ DlgSettingsEditor::DlgSettingsEditor( QWidget* parent )
     QStringList labels; labels << tr("Items");
     ui->displayItems->setHeaderLabels(labels);
     ui->displayItems->header()->hide();
-    for (QVector<QPair<QString, unsigned int> >::Iterator it = d->colormap.begin(); it != d->colormap.end(); ++it) {
+    for (const auto &it: d->colormap)
+    {
         auto item = new QTreeWidgetItem(ui->displayItems);
-        item->setText(0, tr((*it).first.toLatin1()));
+        item->setText(0, tr(it.first.toLatin1()));
```
Comment (chennes) [inline src/Gui/PreferencePages/DlgSettingsEditor.cpp:169]: Consider using a structured binding to unpack these pairs into more sensible variable names, e.g.
```

    for (const auto &[name, index] : d->colormap) {
        auto item = new QTreeWidgetItem(ui->displayItems);
        item->setText(0, tr(name.toLatin1()));
    }
```

Comment (CalligaroV) [reply src/Gui/PreferencePages/DlgSettingsEditor.cpp:169]: Thanks for the suggestion!

I thought that this kind of unpacking was applicable only for Python code but for sure tomorrow I'll give it a try!

File: src/Gui/PreferencePages/DlgSettingsEditor.cpp [hunk 2]
```diff
@@ -280,13 +281,13 @@ void DlgSettingsEditor::loadSettings()
 
     // Restores the color map
     ParameterGrp::handle hGrp = WindowParameter::getDefaultParameter()->GetGroup("Editor");
-    for (QVector<QPair<QString, unsigned int>>::Iterator it = d->colormap.begin();
-         it != d->colormap.end(); ++it) {
-        auto col = static_cast<unsigned long>((*it).second);
-        col = hGrp->GetUnsigned((*it).first.toLatin1(), col);
-        (*it).second = static_cast<unsigned int>(col);
+    for (auto &[textType, textColor]: d->colormap)
```
Comment (chennes) [inline src/Gui/PreferencePages/DlgSettingsEditor.cpp:284]: The linter wants whitespace around the :
```suggestion
    for (auto &[textType, textColor] : d->colormap)
```

File: src/Gui/PreferencePages/DlgSettingsEditor.cpp [hunk 3]
```diff
@@ -332,8 +333,9 @@ void DlgSettingsEditor::resetSettingsToDefaults()
     ParameterGrp::handle hGrp;
     hGrp = WindowParameter::getDefaultParameter()->GetGroup("Editor");
     //reset the parameters in the "Editor" group
-    for (QVector<QPair<QString, unsigned int> >::Iterator it = d->colormap.begin(); it != d->colormap.end(); ++it) {
-        hGrp->RemoveUnsigned((*it).first.toLatin1());
+    for (const auto &[textType, textColor]: d->colormap)
```
Comment (chennes) [inline src/Gui/PreferencePages/DlgSettingsEditor.cpp:336]:
```suggestion
    for (const auto &[textType, textColor] : d->colormap)
```

**Review by chennes — CHANGES_REQUESTED** [review]
Two minor linter issues.

**Pull request merged**